<!DOCTYPE html>
<html>

  <head>
    <meta charset="utf-8">
    <meta http-equiv="X-UA-Compatible" content="IE=edge">
    <title>README for the SILVA v119 reference files</title>
    <meta name="viewport" content="width=device-width">
    <meta name="description" content="A discussion of quantitative isues in microbial ecology">
    <link rel="canonical" href="http://blog.mothur.org/2014/08/08/SILVA-v119-reference-files/">

    <!-- Custom CSS -->
    <link rel="stylesheet" href="/css/main.css">

</head>


    <body>

    <header class="site-header">

  <div class="wrap">

    <a class="site-title" href="/">The mothur blog</a>

    <nav class="site-nav">
      <a href="#" class="menu-icon">
        <svg version="1.1" xmlns="http://www.w3.org/2000/svg" xmlns:xlink="http://www.w3.org/1999/xlink" x="0px" y="0px"
           viewBox="0 0 18 15" enable-background="new 0 0 18 15" xml:space="preserve">
          <path fill="#505050" d="M18,1.484c0,0.82-0.665,1.484-1.484,1.484H1.484C0.665,2.969,0,2.304,0,1.484l0,0C0,0.665,0.665,0,1.484,0
            h15.031C17.335,0,18,0.665,18,1.484L18,1.484z"/>
          <path fill="#505050" d="M18,7.516C18,8.335,17.335,9,16.516,9H1.484C0.665,9,0,8.335,0,7.516l0,0c0-0.82,0.665-1.484,1.484-1.484
            h15.031C17.335,6.031,18,6.696,18,7.516L18,7.516z"/>
          <path fill="#505050" d="M18,13.516C18,14.335,17.335,15,16.516,15H1.484C0.665,15,0,14.335,0,13.516l0,0
            c0-0.82,0.665-1.484,1.484-1.484h15.031C17.335,12.031,18,12.696,18,13.516L18,13.516z"/>
        </svg>
      </a>
      <div class="trigger">
        
          <a class="page-link" href="/about/">About</a>
        
          
        
          
        
      </div>
    </nav>

  </div>

</header>


    <div class="page-content">
      <div class="wrap">
      <div class="post">

  <header class="post-header">
    <h1>README for the SILVA v119 reference files</h1>
    <p class="meta">Aug 8, 2014 • PD Schloss</p>
  </header>

  <article class="post-content">
  <h2 id="curation-of-references">Curation of references</h2>

<h3 id="getting-the-data-in-and-out-of-the-arb-database">Getting the data in and out of the ARB database</h3>
<p>This README file explains how we generated the silva reference files for use with mothur’s classify.seqs and align.seqs commands. I’ll assume that you have a functioning copy of arb installed on your comptuer. For this README we are using version 6.0. First we need to download the database and decompress it. From the command line we do the following:</p>

<div class="highlight"><pre><code class="language-bash" data-lang="bash">wget -N http://www.arb-silva.de/fileadmin/arb_web_db/release_119/ARB_files/SSURef_NR99_119_SILVA_14_07_14_opt.arb.tgz
tar xvzf *arb.tgz
arb SSURef_NR99_119_SILVA_14_07_14_opt.arb</code></pre></div>

<p>This will launch us into the arb environment with the ‘‘Ref NR 99’’ database opened. This database has 534,968 sequences within it that are not more than 99% similar to each other. The release notes for <a href="http://www.arb-silva.de/documentation/release-119/">this database</a> as well as the idea behind the <a href="http://www.arb-silva.de/projects/ssu-ref-nr/">non-redundant database</a> are available from the silva website. Within arb do the following:  </p>

<ol>
  <li>Click the search button</li>
  <li>Set the first search field to ‘ARB_color’ and set it to 1. Click on the equal sign until it indicates not equal (this removes low quality reads and chimeras)</li>
  <li>Click ‘Search’. This yielded 472365 hits</li>
  <li>Click the “Mark Listed Unmark Rest” button</li>
  <li>Close the “Search and Query” box</li>
  <li>Now click on File-&gt;export-&gt;export to external format</li>
  <li>In this box the <code>Export</code> option should be set to <code>marked</code>, <code>Filter</code> to <code>none</code>, and <code>Compression</code> should be set to <code>no</code>.</li>
  <li>In the field for <code>Choose an output file name enter</code> make sure the path has you in the <code>arb_ref_119</code> folder and enter <code>silva.full_v119.fasta</code>.</li>
  <li>Select a format: fasta_mothur.eft. This is a custom formatting file that I have created that includes the sequences accession number and it’s taxonomy across the top line. To create one for you will need to create <code>fasta_mothur.eft</code> in the <code>/opt/local/share/arb/lib/export/</code> folder with the following:</li>
</ol>

<pre><code>SUFFIX          fasta
BEGIN
&gt;*(acc).*(name)\t*(align_ident_slv)\t*(tax_slv);
*(|export_sequence)
</code></pre>

<p>You can now quit arb.  </p>

<h3 id="screening-the-sequences">Screening the sequences</h3>
<p>Now we need to screen the sequences for those that span the 27f and 1492r primer region, have 5 or fewer ambiguous base calls, and that are unique. We’ll also extract the taxonomic information from the header line:</p>

<div class="highlight"><pre><code class="language-bash" data-lang="bash">mothur <span class="s2">&quot;#screen.seqs(fasta=silva.full_v119.fasta, start=1044, end=43116, maxambig=5, processors=8);</span>
<span class="s2">        pcr.seqs(start=1044, end=43116, keepdots=T);</span>
<span class="s2">        degap.seqs();</span>
<span class="s2">        unique.seqs();&quot;</span>
grep <span class="s2">&quot;&gt;&quot;</span> silva.full_v119.good.pcr.ng.unique.fasta <span class="p">|</span> cut -f <span class="m">1</span> <span class="p">|</span> cut -c 2- &gt; silva.full_v119.good.pcr.ng.unique.accnos
mothur <span class="s2">&quot;#get.seqs(fasta=silva.full_v119.good.pcr.fasta, accnos=silva.full_v119.good.pcr.ng.unique.accnos)&quot;</span>

<span class="c">#generate alignment file</span>
mv silva.full_v119.good.pcr.pick.fasta silva.nr_v119.align

<span class="c">#generate taxonomy file</span>
grep <span class="s2">&quot;^&gt;&quot;</span> silva.full_v119.fasta <span class="p">|</span> cut -f 1,3 <span class="p">|</span> cut -c 2- &gt; silva.full_v119.tax.temp</code></pre></div>

<p>The mothur command above does several things. First the screen.seqs command removes sequences that are not full length and have more than 5 ambiguous base calls. Note: this will remove a number of Archaea since the ARB RN reference database lets in shorter (&gt;900 bp) archaeal 16S rRNA gene sequences. Second, pcr.seqs convert any base calls that occur before position 1044 and after 43116 to <code>.</code> to make them only span the region between the 27f and 1492r priming sites. Finally, it is possible that weird things happen in the alignments and so we unalign the sequences (degap.seqs) and identify the unique sequences (unique.seqs). We then convert the resulting fasta file into an accnos file so that we can go back into mothur and pull out the unique sequences from the aligned file (get.seqs).  </p>

<h3 id="formatting-the-taxonomy-files">Formatting the taxonomy files</h3>
<p>Now we want to make sure the taxonomy file is properly formatted for use with mothur.</p>

<div class="highlight"><pre><code class="language-r" data-lang="r">tax <span class="o">&lt;-</span> read.table<span class="p">(</span>file<span class="o">=</span><span class="s">&quot;silva.full_v119.tax.temp&quot;</span><span class="p">,</span> sep<span class="o">=</span><span class="s">&quot;\t&quot;</span><span class="p">)</span>
tax<span class="o">$</span>V2 <span class="o">&lt;-</span> <span class="kp">gsub</span><span class="p">(</span><span class="s">&quot; &quot;</span><span class="p">,</span> <span class="s">&quot;_&quot;</span><span class="p">,</span> tax<span class="o">$</span>V2<span class="p">)</span>  <span class="c1">#convert any spaces to underscores</span>
tax<span class="o">$</span>V2 <span class="o">&lt;-</span> <span class="kp">gsub</span><span class="p">(</span><span class="s">&quot;uncultured;&quot;</span><span class="p">,</span> <span class="s">&quot;&quot;</span><span class="p">,</span> tax<span class="o">$</span>V2<span class="p">)</span>   <span class="c1">#remove any &quot;uncultured&quot; taxa names</span>
tax<span class="o">$</span>V2 <span class="o">&lt;-</span> <span class="kp">paste0</span><span class="p">(</span><span class="s">&quot;Root;&quot;</span><span class="p">,</span> tax<span class="o">$</span>V2<span class="p">)</span>   <span class="c1">#pre-empt all classifications with the Root level.</span>

<span class="c1">#we want to see whether everything has 7 (6) taxonomic levesl (Root to genus)</span>
getDepth <span class="o">&lt;-</span> <span class="kr">function</span><span class="p">(</span>taxonString<span class="p">){</span>
  initial <span class="o">&lt;-</span> <span class="kp">nchar</span><span class="p">(</span>taxonString<span class="p">)</span>
	removed <span class="o">&lt;-</span> <span class="kp">nchar</span><span class="p">(</span><span class="kp">gsub</span><span class="p">(</span><span class="s">&quot;;&quot;</span><span class="p">,</span> <span class="s">&quot;&quot;</span><span class="p">,</span> taxonString<span class="p">))</span>
	<span class="kr">return</span><span class="p">(</span>initial<span class="o">-</span>removed<span class="p">)</span>
<span class="p">}</span>

depth <span class="o">&lt;-</span> getDepth<span class="p">(</span>tax<span class="o">$</span>V2<span class="p">)</span>
bacteria <span class="o">&lt;-</span> <span class="kp">grepl</span><span class="p">(</span><span class="s">&quot;Bacteria;&quot;</span><span class="p">,</span> tax<span class="o">$</span>V2<span class="p">)</span>
archaea <span class="o">&lt;-</span> <span class="kp">grepl</span><span class="p">(</span><span class="s">&quot;Archaea;&quot;</span><span class="p">,</span> tax<span class="o">$</span>V2<span class="p">)</span>
eukarya <span class="o">&lt;-</span> <span class="kp">grepl</span><span class="p">(</span><span class="s">&quot;Eukaryota;&quot;</span><span class="p">,</span> tax<span class="o">$</span>V2<span class="p">)</span>

tax<span class="p">[</span>depth <span class="o">&gt;</span> <span class="m">7</span> <span class="o">&amp;</span> bacteria<span class="p">,]</span>   <span class="c1">#we see that 5 sequences have an 8th level that is actually a repeat of level 7; let&#39;s fix that</span>
tax<span class="p">[</span>depth <span class="o">&gt;</span> <span class="m">7</span> <span class="o">&amp;</span> bacteria<span class="p">,</span><span class="m">2</span><span class="p">]</span> <span class="o">&lt;-</span> <span class="kp">gsub</span><span class="p">(</span><span class="s">&quot;([^;]*);\\1;$&quot;</span><span class="p">,</span> <span class="s">&quot;\\1;&quot;</span><span class="p">,</span> tax<span class="p">[</span>depth <span class="o">&gt;</span> <span class="m">7</span> <span class="o">&amp;</span> bacteria<span class="p">,</span><span class="m">2</span><span class="p">]</span> <span class="p">)</span>
depth <span class="o">&lt;-</span> getDepth<span class="p">(</span>tax<span class="o">$</span>V2<span class="p">)</span>
tax<span class="p">[</span>depth <span class="o">&gt;</span> <span class="m">7</span> <span class="o">&amp;</span> bacteria<span class="p">,]</span> <span class="c1">#good to go</span>
tax<span class="p">[</span>depth <span class="o">&gt;</span> <span class="m">7</span> <span class="o">&amp;</span> archaea<span class="p">,]</span>  <span class="c1">#good to go</span>
tax<span class="p">[</span>depth <span class="o">&gt;</span> <span class="m">7</span> <span class="o">&amp;</span> eukarya<span class="p">,]</span>  <span class="c1">#eh, there&#39;s a lot here - will truncate to the pseudo genus level</span>
tax<span class="p">[</span>depth <span class="o">&gt;</span> <span class="m">7</span> <span class="o">&amp;</span> eukarya<span class="p">,</span><span class="m">2</span><span class="p">]</span> <span class="o">&lt;-</span> <span class="kp">gsub</span><span class="p">(</span><span class="s">&quot;([^;]*;[^;]*;[^;]*;[^;]*;[^;]*;[^;]*;[^;]*;).*&quot;</span><span class="p">,</span> <span class="s">&quot;\\1&quot;</span><span class="p">,</span> tax<span class="p">[</span>depth <span class="o">&gt;</span> <span class="m">7</span> <span class="o">&amp;</span> eukarya<span class="p">,</span><span class="m">2</span><span class="p">])</span>
depth <span class="o">&lt;-</span> getDepth<span class="p">(</span>tax<span class="o">$</span>V2<span class="p">)</span>
tax<span class="p">[</span>depth <span class="o">&gt;</span> <span class="m">7</span> <span class="o">&amp;</span> eukarya<span class="p">,]</span>  <span class="c1">#good to go</span>

write.table<span class="p">(</span>tax<span class="p">,</span> file<span class="o">=</span><span class="s">&quot;silva.full_v119.tax&quot;</span><span class="p">,</span> quote<span class="o">=</span><span class="bp">F</span><span class="p">,</span> row.names<span class="o">=</span><span class="bp">F</span><span class="p">,</span> col.names<span class="o">=</span><span class="bp">F</span><span class="p">)</span></code></pre></div>

<h3 id="building-the-seed-references">Building the SEED references</h3>
<p>Now we want to figure out which sequences are part of the seed. Earlier, when we exported the sequences from ARB, we included the align_ident_slv field from the database in our output. Let’s generate an accnos file that contains the names of the seuqences with 100% to the SEED database and then use mothur to generate SEED fasta and taxonomy files. While we’re at it we’ll also generate the nr_119 taxonomy file as well…:</p>

<div class="highlight"><pre><code class="language-bash" data-lang="bash">grep <span class="s2">&quot;&gt;&quot;</span> silva.nr_v119.align <span class="p">|</span> cut -f 1,2 <span class="p">|</span> grep <span class="s2">&quot;\t100&quot;</span> <span class="p">|</span> cut -f <span class="m">1</span> <span class="p">|</span> cut -c 2- &gt; silva.seed_v119.accnos
mothur <span class="s2">&quot;#get.seqs(fasta=silva.nr_v119.align, taxonomy=silva.full_v119.tax, accnos=silva.seed_v119.accnos)&quot;</span>
mv silva.nr_v119.pick.align silva.seed_v119.align
mv silva.full_v119.pick.tax silva.seed_v119.tax

mothur <span class="s2">&quot;#get.seqs(taxonomy=silva.full_v119.tax, accnos=silva.full_v119.good.pcr.ng.unique.accnos)&quot;</span>
mv silva.full_v119.pick.tax silva.nr_v119.tax</code></pre></div>

<h3 id="taxonomic-representation">Taxonomic representation</h3>
<p>Let’s look to see how many different taxa we have for each taxonomic level within the <code>silva.full_v119.tax</code>, <code>silva.nr_v119.tax</code>, <code>silva.seed_v119.tax</code>:</p>

<div class="highlight"><pre><code class="language-r" data-lang="r">getNumTaxaNames <span class="o">&lt;-</span> <span class="kr">function</span><span class="p">(</span><span class="kp">file</span><span class="p">,</span> kingdom<span class="p">){</span>
  taxonomy <span class="o">&lt;-</span> read.table<span class="p">(</span>file<span class="o">=</span><span class="kp">file</span><span class="p">,</span> row.names<span class="o">=</span><span class="m">1</span><span class="p">)</span>
  sub.tax <span class="o">&lt;-</span> <span class="kp">as.character</span><span class="p">(</span>taxonomy<span class="p">[</span><span class="kp">grepl</span><span class="p">(</span>kingdom<span class="p">,</span> taxonomy<span class="p">[,</span><span class="m">1</span><span class="p">]),])</span>
  
  phyla <span class="o">&lt;-</span> <span class="kp">as.vector</span><span class="p">(</span><span class="kp">levels</span><span class="p">(</span><span class="kp">as.factor</span><span class="p">(</span><span class="kp">gsub</span><span class="p">(</span><span class="s">&quot;[^;]*;[^;]*;([^;]*;).*&quot;</span><span class="p">,</span> <span class="s">&quot;\\1&quot;</span><span class="p">,</span> sub.tax<span class="p">))))</span>
  phyla <span class="o">&lt;-</span> <span class="kp">sum</span><span class="p">(</span><span class="o">!</span><span class="kp">grepl</span><span class="p">(</span><span class="s">&quot;Root&quot;</span><span class="p">,</span> phyla<span class="p">))</span>
  
  class <span class="o">&lt;-</span> <span class="kp">as.vector</span><span class="p">(</span><span class="kp">levels</span><span class="p">(</span><span class="kp">as.factor</span><span class="p">(</span><span class="kp">gsub</span><span class="p">(</span><span class="s">&quot;[^;]*;[^;]*;[^;]*;([^;]*;).*&quot;</span><span class="p">,</span> <span class="s">&quot;\\1&quot;</span><span class="p">,</span> sub.tax<span class="p">))))</span>
  class <span class="o">&lt;-</span> <span class="kp">sum</span><span class="p">(</span><span class="o">!</span><span class="kp">grepl</span><span class="p">(</span><span class="s">&quot;Root&quot;</span><span class="p">,</span> <span class="kp">class</span><span class="p">))</span>
  
  order <span class="o">&lt;-</span> <span class="kp">as.vector</span><span class="p">(</span><span class="kp">levels</span><span class="p">(</span><span class="kp">as.factor</span><span class="p">(</span><span class="kp">gsub</span><span class="p">(</span><span class="s">&quot;[^;]*;[^;]*;[^;]*;[^;]*;([^;]*;).*&quot;</span><span class="p">,</span> <span class="s">&quot;\\1&quot;</span><span class="p">,</span> sub.tax<span class="p">))))</span>
  order <span class="o">&lt;-</span> <span class="kp">sum</span><span class="p">(</span><span class="o">!</span><span class="kp">grepl</span><span class="p">(</span><span class="s">&quot;Root&quot;</span><span class="p">,</span> <span class="kp">order</span><span class="p">))</span>
  
  family <span class="o">&lt;-</span> <span class="kp">as.vector</span><span class="p">(</span><span class="kp">levels</span><span class="p">(</span><span class="kp">as.factor</span><span class="p">(</span><span class="kp">gsub</span><span class="p">(</span><span class="s">&quot;[^;]*;[^;]*;[^;]*;[^;]*;[^;]*;([^;]*;).*&quot;</span><span class="p">,</span> <span class="s">&quot;\\1&quot;</span><span class="p">,</span> sub.tax<span class="p">))))</span>
  family <span class="o">&lt;-</span> <span class="kp">sum</span><span class="p">(</span><span class="o">!</span><span class="kp">grepl</span><span class="p">(</span><span class="s">&quot;Root&quot;</span><span class="p">,</span> family<span class="p">))</span>
  
  genus <span class="o">&lt;-</span> <span class="kp">as.vector</span><span class="p">(</span><span class="kp">levels</span><span class="p">(</span><span class="kp">as.factor</span><span class="p">(</span><span class="kp">gsub</span><span class="p">(</span><span class="s">&quot;[^;]*;[^;]*;[^;]*;[^;]*;[^;]*;[^;]*;([^;]*;).*&quot;</span><span class="p">,</span> <span class="s">&quot;\\1&quot;</span><span class="p">,</span> sub.tax<span class="p">))))</span>
  genus <span class="o">&lt;-</span> <span class="kp">sum</span><span class="p">(</span><span class="o">!</span><span class="kp">grepl</span><span class="p">(</span><span class="s">&quot;Root&quot;</span><span class="p">,</span> genus<span class="p">))</span>

  n.seqs <span class="o">&lt;-</span> <span class="kp">length</span><span class="p">(</span>sub.tax<span class="p">)</span>
  <span class="kr">return</span><span class="p">(</span><span class="kt">c</span><span class="p">(</span>phyla<span class="o">=</span>phyla<span class="p">,</span> class<span class="o">=</span><span class="kp">class</span><span class="p">,</span> order<span class="o">=</span><span class="kp">order</span><span class="p">,</span> family<span class="o">=</span>family<span class="p">,</span> genus<span class="o">=</span>genus<span class="p">,</span> n.seqs<span class="o">=</span>n.seqs<span class="p">))</span>
<span class="p">}</span>

kingdoms <span class="o">&lt;-</span> <span class="kt">c</span><span class="p">(</span><span class="s">&quot;Bacteria&quot;</span><span class="p">,</span> <span class="s">&quot;Archaea&quot;</span><span class="p">,</span> <span class="s">&quot;Eukaryota&quot;</span><span class="p">)</span>
tax.levels <span class="o">&lt;-</span> <span class="kt">c</span><span class="p">(</span><span class="s">&quot;phyla&quot;</span><span class="p">,</span> <span class="s">&quot;class&quot;</span><span class="p">,</span> <span class="s">&quot;order&quot;</span><span class="p">,</span> <span class="s">&quot;family&quot;</span><span class="p">,</span> <span class="s">&quot;genus&quot;</span><span class="p">,</span> <span class="s">&quot;n.seqs&quot;</span><span class="p">)</span>

full.file <span class="o">&lt;-</span> <span class="s">&quot;silva.full_v119.tax&quot;</span>
full.matrix <span class="o">&lt;-</span> <span class="kt">matrix</span><span class="p">(</span><span class="kp">rep</span><span class="p">(</span><span class="m">0</span><span class="p">,</span><span class="m">18</span><span class="p">),</span> nrow<span class="o">=</span><span class="m">3</span><span class="p">)</span>
full.matrix<span class="p">[</span><span class="m">1</span><span class="p">,]</span> <span class="o">&lt;-</span> getNumTaxaNames<span class="p">(</span>full.file<span class="p">,</span> kingdoms<span class="p">[</span><span class="m">1</span><span class="p">])</span>
full.matrix<span class="p">[</span><span class="m">2</span><span class="p">,]</span> <span class="o">&lt;-</span> getNumTaxaNames<span class="p">(</span>full.file<span class="p">,</span> kingdoms<span class="p">[</span><span class="m">2</span><span class="p">])</span>
full.matrix<span class="p">[</span><span class="m">3</span><span class="p">,]</span> <span class="o">&lt;-</span> getNumTaxaNames<span class="p">(</span>full.file<span class="p">,</span> kingdoms<span class="p">[</span><span class="m">3</span><span class="p">])</span>
<span class="kp">rownames</span><span class="p">(</span>full.matrix<span class="p">)</span> <span class="o">&lt;-</span> kingdoms
<span class="kp">colnames</span><span class="p">(</span>full.matrix<span class="p">)</span> <span class="o">&lt;-</span> tax.levels
full.matrix
<span class="c1">#          phyla class order family genus n.seqs</span>
<span class="c1">#Bacteria     61   148   334    631  2281 421951</span>
<span class="c1">#Archaea       8    40    35     75   129  17641</span>
<span class="c1">#Eukaryota    13    30    82    170   709  32773</span>

nr.file <span class="o">&lt;-</span> <span class="s">&quot;silva.nr_v119.tax&quot;</span>
nr.matrix <span class="o">&lt;-</span> <span class="kt">matrix</span><span class="p">(</span><span class="kp">rep</span><span class="p">(</span><span class="m">0</span><span class="p">,</span><span class="m">18</span><span class="p">),</span> nrow<span class="o">=</span><span class="m">3</span><span class="p">)</span>
nr.matrix<span class="p">[</span><span class="m">1</span><span class="p">,]</span> <span class="o">&lt;-</span> getNumTaxaNames<span class="p">(</span>nr.file<span class="p">,</span> kingdoms<span class="p">[</span><span class="m">1</span><span class="p">])</span>
nr.matrix<span class="p">[</span><span class="m">2</span><span class="p">,]</span> <span class="o">&lt;-</span> getNumTaxaNames<span class="p">(</span>nr.file<span class="p">,</span> kingdoms<span class="p">[</span><span class="m">2</span><span class="p">])</span>
nr.matrix<span class="p">[</span><span class="m">3</span><span class="p">,]</span> <span class="o">&lt;-</span> getNumTaxaNames<span class="p">(</span>nr.file<span class="p">,</span> kingdoms<span class="p">[</span><span class="m">3</span><span class="p">])</span>
<span class="kp">rownames</span><span class="p">(</span>nr.matrix<span class="p">)</span> <span class="o">&lt;-</span> kingdoms
<span class="kp">colnames</span><span class="p">(</span>nr.matrix<span class="p">)</span> <span class="o">&lt;-</span> tax.levels
nr.matrix
<span class="c1">#          phyla class order family genus n.seqs</span>
<span class="c1">#Bacteria     59   144   318    594  2174 137879</span>
<span class="c1">#Archaea       7    27    30     63   123   3155</span>
<span class="c1">#Eukaryota     9    20    66    119   547  12273</span>

seed.file <span class="o">&lt;-</span> <span class="s">&quot;silva.seed_v119.tax&quot;</span>
seed.matrix <span class="o">&lt;-</span> <span class="kt">matrix</span><span class="p">(</span><span class="kp">rep</span><span class="p">(</span><span class="m">0</span><span class="p">,</span><span class="m">18</span><span class="p">),</span> nrow<span class="o">=</span><span class="m">3</span><span class="p">)</span>
seed.matrix<span class="p">[</span><span class="m">1</span><span class="p">,]</span> <span class="o">&lt;-</span> getNumTaxaNames<span class="p">(</span>seed.file<span class="p">,</span> kingdoms<span class="p">[</span><span class="m">1</span><span class="p">])</span>
seed.matrix<span class="p">[</span><span class="m">2</span><span class="p">,]</span> <span class="o">&lt;-</span> getNumTaxaNames<span class="p">(</span>seed.file<span class="p">,</span> kingdoms<span class="p">[</span><span class="m">2</span><span class="p">])</span>
seed.matrix<span class="p">[</span><span class="m">3</span><span class="p">,]</span> <span class="o">&lt;-</span> getNumTaxaNames<span class="p">(</span>seed.file<span class="p">,</span> kingdoms<span class="p">[</span><span class="m">3</span><span class="p">])</span>
<span class="kp">rownames</span><span class="p">(</span>seed.matrix<span class="p">)</span> <span class="o">&lt;-</span> kingdoms
<span class="kp">colnames</span><span class="p">(</span>seed.matrix<span class="p">)</span> <span class="o">&lt;-</span> tax.levels
seed.matrix
<span class="c1">#          phyla class order family genus n.seqs</span>
<span class="c1">#Bacteria     49    95   189    382  1545  12244</span>
<span class="c1">#Archaea       4    14    18     31    66    207</span>
<span class="c1">#Eukaryota     7    14    23     45   127   2558</span>

nr.matrix <span class="o">/</span> full.matrix
<span class="c1">#              phyla     class     order    family     genus</span>
<span class="c1">#Bacteria  0.9672131 0.9729730 0.9520958 0.9413629 0.9530907</span>
<span class="c1">#Archaea   0.8750000 0.6750000 0.8571429 0.8400000 0.9534884</span>
<span class="c1">#Eukaryota 0.6923077 0.6666667 0.8048780 0.7000000 0.7715092</span>

seed.matrix <span class="o">/</span> full.matrix
<span class="c1">#              phyla     class     order    family     genus</span>
<span class="c1">#Bacteria  0.8032787 0.6418919 0.5658683 0.6053883 0.6773345</span>
<span class="c1">#Archaea   0.5000000 0.3500000 0.5142857 0.4133333 0.5116279</span>
<span class="c1">#Eukaryota 0.5384615 0.4666667 0.2804878 0.2647059 0.1791255</span></code></pre></div>

<p>We see that our full-length database retains a signigicant majority of the taxa that were in the original NR database. The Archaea take a beating. If you are interested in analyzing the Eukaryota, I would suggest duplicating my efforts here but modify the screen.seqs and pcr.seqs steps to target your region of interest.  </p>

<p>Finally, we want to compress the resulting alignment and this README file into the full length and SEED archives:</p>

<div class="highlight"><pre><code class="language-bash" data-lang="bash">tar cvzf silva.nr_v119.tgz silva.nr_v119.tax silva.nr_v119.align README.*
tar cvzf silva.seed_v119.tgz silva.seed_v119.tax silva.seed_v119.align README.*</code></pre></div>

<h2 id="application">Application</h2>
<p>So… which to use for what application? If you have the RAM, I’d suggest using <code>silva.nr_v119.align</code> in <code>align.seqs</code>. It took about 10 minutes to read in the database file and a minute or so to align a 1000 full-length sequences. Here is an example workflow for use within mothur that will get you the V4 region of the 16S rRNA gene:</p>

<div class="highlight"><pre><code class="language-bash" data-lang="bash">mothur <span class="s2">&quot;#pcr.seqs(fasta=silva.nr_v119.align, start=11894, end=25319, keepdots=F, processors=8);</span>
<span class="s2">        unique.seqs()&quot;</span></code></pre></div>

<p>This will get you 104,711 unique sequences to then align against (meh.). Other tricks to consider would be to use <code>get.lineage</code> to pull out the reference sequences that are from the Bacteria, this will probably only reduce the size of the database by ~10%. You could also try using <code>filter.seqs</code> with vertical=T; however, that might be problematic if there are insertions in your sequences (can’t know <em>a priori</em>). It’s likely that you can just use the <code>silva.seed_v119.align</code> reference for aligning. For classifying sequences, I would strongly recommend using the <code>silva.nr_v119.align</code> and <code>silva.nr_v119.tax</code> references after running pcr.seqs on <code>silva.nr_v119.align</code>. I probably wouldn’t advise using <code>unique.seqs</code> on the output.</p>

  </article>

</div>
      </div>
    </div>

    <footer class="site-footer">

  <div class="wrap">

    <h2 class="footer-heading">The mothur blog</h2>

    <div class="footer-col-1 column">
      <ul>
        <li>The mothur blog</li>
        <li><a href="mailto:pschloss@umich.edu">pschloss@umich.edu</a></li>
      </ul>
    </div>

    <div class="footer-col-2 column">
      <ul>
        <li>
          <a href="https://github.com/pschloss">
            <span class="icon github">
              <svg version="1.1" class="github-icon-svg" xmlns="http://www.w3.org/2000/svg" xmlns:xlink="http://www.w3.org/1999/xlink" x="0px" y="0px"
                 viewBox="0 0 16 16" enable-background="new 0 0 16 16" xml:space="preserve">
                <path fill-rule="evenodd" clip-rule="evenodd" fill="#C2C2C2" d="M7.999,0.431c-4.285,0-7.76,3.474-7.76,7.761
                c0,3.428,2.223,6.337,5.307,7.363c0.388,0.071,0.53-0.168,0.53-0.374c0-0.184-0.007-0.672-0.01-1.32
                c-2.159,0.469-2.614-1.04-2.614-1.04c-0.353-0.896-0.862-1.135-0.862-1.135c-0.705-0.481,0.053-0.472,0.053-0.472
                c0.779,0.055,1.189,0.8,1.189,0.8c0.692,1.186,1.816,0.843,2.258,0.645c0.071-0.502,0.271-0.843,0.493-1.037
                C4.86,11.425,3.049,10.76,3.049,7.786c0-0.847,0.302-1.54,0.799-2.082C3.768,5.507,3.501,4.718,3.924,3.65
                c0,0,0.652-0.209,2.134,0.796C6.677,4.273,7.34,4.187,8,4.184c0.659,0.003,1.323,0.089,1.943,0.261
                c1.482-1.004,2.132-0.796,2.132-0.796c0.423,1.068,0.157,1.857,0.077,2.054c0.497,0.542,0.798,1.235,0.798,2.082
                c0,2.981-1.814,3.637-3.543,3.829c0.279,0.24,0.527,0.713,0.527,1.437c0,1.037-0.01,1.874-0.01,2.129
                c0,0.208,0.14,0.449,0.534,0.373c3.081-1.028,5.302-3.935,5.302-7.362C15.76,3.906,12.285,0.431,7.999,0.431z"/>
              </svg>
            </span>
            <span class="username">pschloss</span>
          </a>
        </li>
        <li>
          <a href="https://twitter.com/patschloss">
            <span class="icon twitter">
              <svg version="1.1" class="twitter-icon-svg" xmlns="http://www.w3.org/2000/svg" xmlns:xlink="http://www.w3.org/1999/xlink" x="0px" y="0px"
                 viewBox="0 0 16 16" enable-background="new 0 0 16 16" xml:space="preserve">
                <path fill="#C2C2C2" d="M15.969,3.058c-0.586,0.26-1.217,0.436-1.878,0.515c0.675-0.405,1.194-1.045,1.438-1.809
                c-0.632,0.375-1.332,0.647-2.076,0.793c-0.596-0.636-1.446-1.033-2.387-1.033c-1.806,0-3.27,1.464-3.27,3.27
                c0,0.256,0.029,0.506,0.085,0.745C5.163,5.404,2.753,4.102,1.14,2.124C0.859,2.607,0.698,3.168,0.698,3.767
                c0,1.134,0.577,2.135,1.455,2.722C1.616,6.472,1.112,6.325,0.671,6.08c0,0.014,0,0.027,0,0.041c0,1.584,1.127,2.906,2.623,3.206
                C3.02,9.402,2.731,9.442,2.433,9.442c-0.211,0-0.416-0.021-0.615-0.059c0.416,1.299,1.624,2.245,3.055,2.271
                c-1.119,0.877-2.529,1.4-4.061,1.4c-0.264,0-0.524-0.015-0.78-0.046c1.447,0.928,3.166,1.469,5.013,1.469
                c6.015,0,9.304-4.983,9.304-9.304c0-0.142-0.003-0.283-0.009-0.423C14.976,4.29,15.531,3.714,15.969,3.058z"/>
              </svg>
            </span>
            <span class="username">patschloss</span>
          </a>
        </li>
      </ul>
    </div>

    <div class="footer-col-3 column">
      <p class="text">A discussion of quantitative isues in microbial ecology</p>
    </div>

  </div>

</footer>


    </body>
</html>